modules/management-console-agent/src/main/java/org/apache/ignite/agent/processor/ClusterInfoProcessor.java

```diff
@@ -129,25 +129,23 @@ void sendClusterInfo(Event evt) {
         if (log.isDebugEnabled())
             log.debug("Sending cluster info to Control Center");
 
-        ClusterInfo clusterInfo = createClusterInfo();
-
-        populateClusterInfo(clusterInfo);
-
-        mgr.send(buildClusterDest(cluster.id()), clusterInfo);
+        mgr.send(buildClusterDest(cluster.id()), buildClusterInfo());
     }
 
     /**
      * @return Create cluster info.
      */
     protected ClusterInfo createClusterInfo() {
```

Comment:
It seems that `buildClusterInfo()` should be made `protected` and overloaded in the ancestor, not `createClusterInfo`.

Reply:
In this case we will copy-paste  follow part of code:

```java
.setId(cluster.id())
            .setTag(cluster.tag())
            .setActive(cluster.active())
            .setPersistenceEnabled(CU.isPersistenceEnabled(ctx.config()))
            .setBaselineParameters(
                new BaselineInfo(
                    cluster.isBaselineAutoAdjustEnabled(),
                    cluster.baselineAutoAdjustTimeout()
                )
            )
            .setSecure(ctx.authentication().enabled() || ctx.security().enabled())
            .setFeatures(getClusterFeatures(ctx, ctx.cluster().get().nodes()));
```

Reply:
Just use `super.buildClusterInfo()` to construct an instance of `ClusterInfo` in the ancestor.

Reply:
We can't do this because if we call `super.buildClusterInfo()` we construct `ClusterInfo` which doesn't contains additional fields. In this case we need to copy all it's fields to new `GridGainClusterInfo` which may leads to unexpected errors if we add some field to `ClusterInfo` and don't copy it to `GridGainClusterInfo`

Reply:
Oh, yeah, you are right. Sorry, I missed that.
But it's still hard to tell the difference between these two methods. What's the difference between "create" and "build"?
One method creates an empty instance of `ClusterInfo`, another one creates **AND** populates it. But `GridGainClusterInfoProcessor#createClusterInfo` actually does both of those things, while it's supposed to do only creating, judging by the name.

I like the previous design when you had two separate methods for creating and populating the ClusterInfo instances more. I suggest falling back to it.

Reply:
Partial revert